### PR TITLE
 manifest docs

### PR DESCRIPTION
**Pull request by NitiKaur**

### CODE REVIEW

**Comment by mattytrentini** [comment]
See #7601

File: docs/reference/manifest.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        options.defaults(standard_features=True)
+        # freeze minimal modules.
+        if options.standard_features:
+            # freeze standard modules.
+        if options.extra_features:
+            # freeze extra modules.
+         
+The MicroPython manifest files supports variousmethods which are:
+
+* yasmarang_randbelow : returns an unsigned integer below the given argument
```
Comment (dpgeorge) [inline docs/reference/manifest.rst:23]: this should not be here

File: docs/reference/manifest.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                         
+                        Syntax - STATIC uint32_t yasmarang_randbelow(uint32_t n)
+                                  
+                                 (Here, n shouldn't be 0)
+
+* yasmarang_float : returns a number in the range [0..1) using Yasmarang to
```
Comment (dpgeorge) [inline docs/reference/manifest.rst:29]: this should net be here

File: docs/reference/manifest.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+* freeze_as_str(path) : Freeze the given `path` and all .py scripts within it
+                        as a string, which will be compiled upon import.
+
+* freeze_as_mpy(path, script=None, opt=0) : Freeze the input (see above) by
+                                            first compiling the .py scripts 
+                                            to .mpy files, then freezing the 
+                                            resulting.mpy files.
+
+def freeze_mpy(path, script=None, opt=0): Freeze the input (see above), which
```
Comment (dpgeorge) [inline docs/reference/manifest.rst:42]: remove the "def" keyword

**Review by dpgeorge — CHANGES_REQUESTED** [review]

**Comment by dpgeorge** [comment]
Would be good to include a few examples, eg from `ports/rp2/boards/manifest.py`.

**Comment by dpgeorge** [comment]
The public functions are:
- `include`
- `freeze`
- `freeze_as_str`
- `freeze_as_mpy`
- `freeze_mpy`

**Comment by pidou46** [comment]
Maybe a word to explain what the manifest file actually is, what is the aim, before explain how it works ?
Except if it's explain elsewhere.

**Comment by dpgeorge** [comment]
Merged in e87b2e8bfa71a09f0fe585fc0127e9fa42b68f19 with layout improvements and an examples section.

**Pull request closed without merging**